### PR TITLE
Bad double bond in linear group

### LINKED ISSUE #6027: Bad double bond drawn for collinear atoms
**Describe the bug**
When drawing a double bond on an exactly collinear bond, one end of the bond is badly wrong.

**To Reproduce**
```
collinear_molblock = """
  MJ201100                      

 15 16  0  0  0  0  0  0  0  0999 V2000
   -2.6052   -0.0931    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -2.6052    0.7318    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -3.3197    1.1444    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0
   -4.0346    0.7318    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -4.0347   -0.0931    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -3.3197   -0.5055    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -3.3197    1.9693    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -3.3197    2.7943    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.8908   -0.5055    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.1764    0.7318    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.1764   -0.0931    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.4620   -0.5055    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.2523   -0.0931    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.2524    0.7318    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.4620    1.1444    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0  0  0  0
  2  3  1  0  0  0  0
  5  6  1  0  0  0  0
  6  1  1  0  0  0  0
  3  7  1  0  0  0  0
  7  8  2  0  0  0  0
 13 14  1  0  0  0  0
 14 15  1  0  0  0  0
 10 11  1  0  0  0  0
 15 10  1  0  0  0  0
 11 12  1  0  0  0  0
 12 13  1  0  0  0  0
  3  4  1  0  0  0  0
  4  5  1  0  0  0  0
  9  1  1  0  0  0  0
 11  9  1  0  0  0  0
M  END
"""
collinear_mol = Chem.MolFromMolBlock(collinear_molblock)
collinear_mol
```

**Expected behavior**
A double bond drawn correctly.

**Screenshots**
image

**Configuration (please complete the following information):**
 - RDKit version: 2022.09.3, master
 - OS: macOS Monterey 13.2, Apple Silicon
 - Python version (if relevant): 3.10.4
 - Are you using conda? mamba
 - If you are using conda, which channel did you install the rdkit from? conda-forge
 - If you are not using conda: how did you install the RDKit? Also built from master branch.

**Additional context**
Discovered by @ptosco 
Fix imminent.

### PR DESCRIPTION
**Pull request by DavidACosgrove**
#### Reference Issue
Fixes #6027

#### What does this implement/fix? Explain your changes.
The linear double bond wasn't being drawn correctly.  I have adapted the Allene-drawing code to deal with this case also. 

#### Any other comments?
This means that the double bond straddles the single bond incident on it, rather than having the second bond alongside the first which is linear to the single bond.  It was easier to do it this way, and it's already a bad 2D picture because it should be a cranked double bond, so I think it's ok.  If anyone objects, it could be done to be more consistent with the way other double bonds are drawn, but not as easily.

### CODE REVIEW

File: Code/GraphMol/MolDraw2D/DrawMol.h [hunk 1]
```diff
@@ -304,7 +304,13 @@ class DrawMol {
...
 void centerMolForDrawing(RWMol &mol, int confId = 1);
 void prepareStereoGroups(RWMol &mol);
-bool isLinearAtom(const Atom &atom, const std::vector<Point2D> &atCds);
+// Check if bonds off the atom are linear, if the atom is of degree 2.
+// If sameBondTypes is true, the 2 bonds off the atom must be of the same
+// type, if false they can be different types.  If they're the same type,
+// we show the atom symbol so it doesn't look like 1 long bond but that's
+// not necessary if they're different types.
+bool isLinearAtom(const Atom &atom, const std::vector<Point2D> &atCds,
+                  bool sameBondTypes = true);
```
Comment (ptosco) [inline Code/GraphMol/MolDraw2D/DrawMol.h:313]:
```suggestion
                  bool checkSameBondType = true);
```

File: Code/GraphMol/MolDraw2D/DrawMol.cpp [hunk 2]
```diff
@@ -3347,7 +3359,8 @@ void prepareStereoGroups(RWMol &mol) {
 }
 
 // ****************************************************************************
-bool isLinearAtom(const Atom &atom, const std::vector<Point2D> &atCds) {
+bool isLinearAtom(const Atom &atom, const std::vector<Point2D> &atCds,
+                  bool sameBondTypes) {
```
Comment (ptosco) [inline Code/GraphMol/MolDraw2D/DrawMol.cpp:3363]:
```suggestion
                  bool checkSameBondType) {
```

File: Code/GraphMol/MolDraw2D/DrawMol.cpp [hunk 3]
```diff
@@ -3361,7 +3374,12 @@ bool isLinearAtom(const Atom &atom, const std::vector<Point2D> &atCds) {
       bts[i] = mol.getBondBetweenAtoms(atom.getIdx(), nbr)->getBondType();
       ++i;
     }
-    return (bts[0] == bts[1] && bond_vecs[0].dotProduct(bond_vecs[1]) < -0.95);
+    if (sameBondTypes) {
+      return (bts[0] == bts[1] &&
+              bond_vecs[0].dotProduct(bond_vecs[1]) < -0.95);
+    } else {
+      return (bond_vecs[0].dotProduct(bond_vecs[1]) < -0.95);
+    }
```
Comment (ptosco) [inline Code/GraphMol/MolDraw2D/DrawMol.cpp:3382]:
```suggestion
    bool eitherSameBondTypeOrDontCare = !checkSameBondType || bts[0] == bts[1];
    return eitherSameBondTypeOrDontCare &&
              bond_vecs[0].dotProduct(bond_vecs[1]) < -0.95);
```

**Review by ptosco — APPROVED** [review]
LGTM, made a few non-binding suggestions.
Many thanks @DavidACosgrove for the super-quick fix!

**Review by greglandrum — APPROVED** [review]
The changes look fine. The changes to the arguments of a function in the public API may have consequences, but I'll raise that in a separate comment.

**Comment by greglandrum** [comment]
This changes the arguments to `isLinearAtom()`, which is in the public API.
That's not something we normally do with a patch release, so that would lead me to mark the bug (and this PR) for the 2023.03.1 release.
There's an alternative  - I can do a version of this patch just for the release branch which adds a new local function instead of changing the args to the global function - but it's work, so I'd rather avoid doing it unless this bug seems important.

I don't have a good sense of how important it is, @ptosco, @DavidACosgrove : what's your take?

**Comment by DavidACosgrove** [comment]
Sorry, I hadn’t spotted it was in the public API. It seemed like the easiest way of solving the problem, but it’s not the only way. I’m wondering why I even had that function as public. 
The bug itself seems fairly trivial to me- it’s a very badly drawn molecule. I’d be interested to know how it came up, even.  Since there’s a merge clash to deal with in any case, I’ll take a look at it tomorrow and see if it can be dealt with in a way that doesn’t change the API.

**Comment by ptosco** [comment]
I agree the molecule is really badly drawn. It is not a forged case, though - I found a couple of those in the real world :-(. So I think it would be good to include it in a patch release as the rendering looks quite bad (I mean, even worse than it should given the poor quality coordinates).

**Comment by DavidACosgrove** [comment]
Ok, I’ll see if I can find a better fix tomorrow.

**Comment by DavidACosgrove** [comment]
I've changed it so it doesn't need the change to isLinearAtom.  I don't know why I didn't do it this way to start with.  Tunnel vision set in, no doubt.

**Review by greglandrum — APPROVED** [review]
LGTM

**Comment by greglandrum** [comment]
> I've changed it so it doesn't need the change to isLinearAtom. I don't know why I didn't do it this way to start with. Tunnel vision set in, no doubt.

Thanks for making that change!

**Pull request merged**